server/services/caseloadService.ts

```diff
@@ -203,13 +203,13 @@ export default class CaseloadService {
       .filter(offender => !offender.nomisRecord.paroleEligibilityDate)
       .filter(offender => offender.nomisRecord.legalStatus !== 'DEAD')
       .filter(offender => !offender.nomisRecord.indeterminateSentence)
-      .filter(offender => offender.nomisRecord.conditionalReleaseDate)
-      // TODO: Following filter rule can be removed after 18th April 2022
-      .filter(offender =>
-        moment(offender.nomisRecord.conditionalReleaseDate, 'YYYY-MM-DD').isSameOrAfter(
-          moment('2022-04-18', 'YYYY-MM-DD'),
-          'day'
-        )
+      // TODO - Check with Jon on rules for filtering recalls
+      // .filter(offender => !offender.nomisRecord.recall)
+      .filter(
+        offender =>
+          offender.nomisRecord.conditionalReleaseDate ||
+          offender.nomisRecord.releaseDate ||
+          offender.nomisRecord.confirmedReleaseDate
```

Comment:
Can you check `licenceService` and `datesChangedEventHandler`? `confirmedReleaseDate` isnt used in these places, and maybe it should be

Reply:
I'll look when I get back in next hour.

Reply:
It looks fine - the datesChangedEventHandler receives either a `nomisId` or a `bookingId` (which it looks up the nomisId), then gets the matching, in-progress (creation only) licences by `nomisId` and then sets the licence to in-progress, pulls the sentence dates from prison-api and updates the dates on the licence (conditionalReleaseDate and actualReleaseDate).

Reply:
One improvement might be to use the `sentenceDetail.releaseDate` IF there is no `conditionalReleaseDate` present in the setence detail.

Reply:
Or even if there is - and do the same thing that the licenceStartDate is set ti - use releaseDate if present, otherwise conditionalReleaseOverrideDate, and otherwise conditionalReleaseDate.